lib/Fencer/Types.hs

```diff
@@ -150,6 +160,14 @@ data DescriptorDefinition = DescriptorDefinition
     }
     deriving stock (Eq, Show)
 
+instance HasDescriptors DomainDefinition where
+  descriptorsOf = domainDefinitionDescriptors
+
+instance HasDescriptors DescriptorDefinition where
+  descriptorsOf d = case descriptorDefinitionDescriptors d of
+    Nothing -> []
+    Just ds -> ds
```

Comment:
Nit: I'd write it as `fromMaybe [] $ descriptorDefinitionDescriptors d`

Reply:
Thank you once again! Fixed.